### PR TITLE
Fix prost proto packages not sanitizing to valid module names

### PR DESCRIPTION
**Pull request by UebelAndre**
This normalizes module names to be consistently written as lowercase. When given extravagant package names `prost-build` will snake_case package names `PackageX.FooBar` would become `package_x.foo_bar`. This change contains two commits where the first solves a bug where the extern paths that the process wrapper generated did not match the the paths `prost-build` would render. Sanitizing these to lowercase is the simplest fix. The second commit contains a change to add snake_case support by bringing in the same library `prost-build` uses to do the same logic but is a more complicated implementation. I personally lean toward the simpler solution but leave it up to my reviewers.

### CODE REVIEW

**Comment by UebelAndre** [comment]
cc @freeformstu curious on your thoughts if this continues to work for your needs too.

**Comment by freeformstu** [comment]
I think this looks great. I'm fine with the additional dependency, that doesn't bother me, and it makes it more correct.

**Comment by UebelAndre** [comment]
> TLDR: Should commit 2 be dropped in favor of the simpler implementation?
> 
>     * Commit 1 - Sanitize rust module names to lowercase
> 
>     * Commit 2 - Bring in a new dependency to sanitize to snake case the way `prost-build` does.

Also curious if @hlopko has thoughts here. How would you expect the package names to be normalized and do you think the added complexity should be added to match the way Prost does things?

**Review by illicitonion — APPROVED** [review]
Thanks!

I would lean towards faithfully minicing `prost-build`'s behaviour as the second commit does.
